Pill Component (#232)
LifeSG/react-design-system: stories/pill/pill.stories.mdx

import { Canvas, Meta, Preview, Story } from "@storybook/addon-docs";
import { PlusCircleFillIcon } from "@lifesg/react-icons/plus-circle-fill";
import { PlaceholderIcon } from "@lifesg/react-icons/placeholder";

import { Pill } from "src/pill";
import { Secondary, Heading3, Title } from "../storybook-common";
import { Grid, Label, Row, Wrapper } from "./doc-elements";
import { PropsTable } from "./props-table";

<Meta title="Modules/Pill" component={Pill} />

<Title>Pill</Title>

<Secondary>Overview</Secondary>

A status pill component.

Example of usage: As a status indicator on current flow state such as `Pending Approval`, `Approved`, `Rejected` and etc.

```tsx
import { Pill } from "@lifesg/react-design-system/pill";
```

> The `Pill` component comes in 2 type, a `solid` and `outline`.

Reviewer: No need for this. To mention this in the variations

<Canvas>
    <Story name="Pill">
        <Wrapper>
            <Pill type="outline" colorType="blue">

>                 Pending Approval

Reviewer: ```suggestion
                Outline style pill
```

            </Pill>
            <br />

>             <Pill type="outline" colorType="green">

Reviewer: Not needed

                Approved
            </Pill>
            <br />
            <Pill type="solid" colorType="red">

>                 Rejected

Reviewer: ```suggestion
                Solid style pill
```

            </Pill>
        </Wrapper>
    </Story>
</Canvas>

> Example of `Pill` usage with icon.

Reviewer: Add a `Heading3` titled "Pill with icon" and this description is no longer needed

<Canvas>
    <Story name="With Icon">
        <Wrapper>
            <Pill
                type="outline"
                icon={<PlusCircleFillIcon />}
                data-testid="first-example"
            >
                Status Label
            </Pill>
        </Wrapper>
        <br />
        <Wrapper>
            <Pill
                type="solid"
                colorType="blue"
                icon={<PlaceholderIcon />}
                data-testid="second-example"
            >
                Status Label
            </Pill>
        </Wrapper>
    </Story>
</Canvas>

<Heading3>Variations</Heading3>

> The `Pill` component comes in 6 colorType, a `black`, `grey`, `green`, `yellow`, `red`, and `blue` color.

Reviewer: ```suggestion
The `Pill` comes in 2 display formats, `outline` and `solid` and there are 6 color variations that can be denoted using the `colorType` prop.
```

> Color `black` will be the default colorType.

Reviewer: Not needed. The props table already indicated

                    </Pill>
                    <Pill type="outline" colorType="red">
                        Label
                    </Pill>
                    <Pill type="outline" colorType="blue">
                        Label
                    </Pill>
                </Row>
                <Row>
                    <Label>Solid</Label>
                    <Pill type="solid" colorType="black">
                        Label
                    </Pill>
                    <Pill type="solid" colorType="grey">
                        Label
                    </Pill>
                    <Pill type="solid" colorType="green">
                        Label
                    </Pill>
                    <Pill type="solid" colorType="yellow">
                        Label
                    </Pill>
                    <Pill type="solid" colorType="red">
                        Label
                    </Pill>
                    <Pill type="solid" colorType="blue">
                        Label
                    </Pill>
                </Row>
            </Grid>
        </StoryContainer>
    </Story>
</Preview>

<Secondary>Component API</Secondary>

> This component also inherits the [HTMLDivAttributes](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDivElement)

Reviewer: To add this within the Props table instead. Refer to `Button` stories
